### DISCUSSION TITLE
Export mapping with related fields

### DISCUSSION THREAD

**Discussion by phiff** [forcedotcom/dataloader#854] [Q&A] [answered]
I'm having problems creating a export mapping file where the fields are from related objects.

Fx

Id=Id
INetCaseId__c=INetCaseId

works fine, but when i try either

BusinessCommitmentSummary__r.LoansCreditsGuaranteesRammerSumCur__c=LoansCreditsGuarRammerSumCur
or
BusinessCommitmentSummary__r\:LoansCreditsGuaranteesRammerSumCur__c=LoansCreditsGuarRammerSumCur

I end up with an empty CSV - no data exported

**Comment by ashitsalesforce** [comment, marked as answer]
Hi @phiff , can you share your SoQL query and `process.mappingFile` property in your config.properties file? 

Here is my org's setup:
2 custom sobjects:
`childCustom__c` sobject:
   - field `parentCustomName__c` is a Lookup field pointing to `parentCustom__c`
   
`parentCustom__c` sobject:
   - Text field `field1__c`

I created the following mapping:
```
name=childname
ParentCustomName__r.field1__c=parentField1
```

I ran the following query:
```
SELECT name, ParentCustomName__r.field1__c
FROM ChildCustom__c  where id='xxxxxxxxxxxxxxxxx'
```

Here is the result in a csv file:
```
"PARENTFIELD1","CHILDNAME"
"More parent","c1"
```

The csv file shows that the field mapping is correctly applied.

**Reply by phiff** [reply]
I'm not quit sure what my problem was, but now it is working.
I double checked my mappings file and found no errors..........but there must have been, because I have now recreated the mapping and it's working just fine.

Sorry for the inconvenience.

Kind Regards
Allan Pheiffer

**Reply by ashitsalesforce** [reply]
Hi @phiff , good to hear that it is working fine. Perhaps something like the following occurred if you were using Data Loader in UI mode:
- Set the mapping `BusinessCommitmentSummary__r:LoansCreditsGuaranteesRammerSumCur__c=LoansCreditsGuarRammerSumCur` before launching Data Loader in UI mode.
- Ran the query but did not see mapping work in the query results csv because the mapping contains : character to separate sobject name from the field name.
- Changed the mapping to `BusinessCommitmentSummary__r.LoansCreditsGuaranteesRammerSumCur__c=LoansCreditsGuarRammerSumCur` without quitting Data Loader. That did not work because Data Loader does not check fo changes to the mapping when it is running. It is working now because you restarted data loader.
